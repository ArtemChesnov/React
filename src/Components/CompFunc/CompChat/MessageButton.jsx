import React from 'react';

export const MessageButton = (props) => {
  return (
    <button onClick={props.click} className="message-button">
      Отправить
    </button>
  );
};
