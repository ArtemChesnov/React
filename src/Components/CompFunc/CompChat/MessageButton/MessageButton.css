.message-button {
    border: none;
    border-radius: 15px;
    padding: 20px 30px;
    cursor: pointer;
    background-color: rgb(212, 212, 212);
    transition: 0.1s ease-in;
    right: 0;
}

.message-button:hover {
    background-color: rgb(184, 178, 178);
}

.message-button:active {
    background-color: rgb(10, 10, 10);
    color: #eee;
}

.message-button:disabled {
    background-color: rgb(46, 34, 34);
    color: #eee;
}