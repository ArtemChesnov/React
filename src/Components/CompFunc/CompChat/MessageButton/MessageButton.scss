.chat-message-button {
    background-color: rgba(255, 255, 255, 0.4);
    padding: 5px 0 5px 0;
    font-size: 0.9em;
    color: rgba(255, 255, 255, 1);
    cursor: pointer;
    border: none;
    margin: 5px 0 0 0;
    font-family: BlinkMacSystemFont,
        "Segoe UI",
        "Roboto",
        "Oxygen",
        "Ubuntu",
        "Cantarell",
        "Fira Sans",
        "Droid Sans",
        "Helvetica Neue",
        sans-serif;
    -webkit-font-smoothing: antialiased;
}

.chat-message-button:hover {
    background-color: rgba(255, 255, 255, 0.1);
}

.chat-message-button:active {
    background-color: rgba(255, 255, 255, 0.5);
}

.chat-message-button:disabled {
    background-color: rgba(255, 255, 255, 0.01);
}