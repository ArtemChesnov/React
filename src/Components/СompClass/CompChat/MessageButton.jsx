import React, { Component } from 'react';

export class MessageButton extends Component {
  render() {
    return (
      <button onClick={this.props.click} className="message-button">
        Отправить
      </button>
    );
  }
}
