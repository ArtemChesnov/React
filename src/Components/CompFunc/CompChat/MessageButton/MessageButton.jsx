import React from 'react';
import './MessageButton.css';

export const MessageButton = ({ disabled, click }) => {
  return (
    <button
      onClick={click}
      disabled={disabled}
      className="message-button"
      id="button"
      type="submit"
    >
      Отправить
    </button>
  );
};
