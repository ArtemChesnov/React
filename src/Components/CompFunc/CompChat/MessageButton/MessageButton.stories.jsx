import React from 'react';

import { MessageButton } from './MessageButton';

export default {
  title: 'Components/MessageButton',
  component: MessageButton,
  argTypes: {
    click: { action: 'click' },
  },
};

const Template = (args) => <MessageButton {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  disabled: false,
};

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
};
