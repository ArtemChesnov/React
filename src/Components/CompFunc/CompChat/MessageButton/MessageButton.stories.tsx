import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { MessageButton } from './MessageButton';

export default {
  title: 'Components/MessageButton',
  component: MessageButton,
  argTypes: {
    onClick: { action: 'click' },
  },
} as ComponentMeta<typeof MessageButton>;

const Template: ComponentStory<typeof MessageButton> = (args) => (
  <MessageButton {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  disabled: false,
};

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
};
