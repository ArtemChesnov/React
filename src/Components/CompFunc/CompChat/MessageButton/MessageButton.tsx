import React, { FC } from 'react';
import './MessageButton.scss';

interface ButtonProps {
  disabled: boolean;
  onClick?: () => void;
}

export const MessageButton: FC<ButtonProps> = ({ disabled, onClick }) => {
  return (
    <button
      disabled={disabled}
      className="chat-message-button"
      id="button"
      type="submit"
      onClick={onClick}
    >
      Отправить
    </button>
  );
};
