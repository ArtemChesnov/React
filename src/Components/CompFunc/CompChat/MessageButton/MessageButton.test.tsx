import React from 'react';
import { MessageButton } from './MessageButton';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { waitFor } from '@storybook/testing-library';

describe('MessageButton', () => {
  it('render component', () => {
    render(<MessageButton disabled={false} />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<MessageButton disabled={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component with text', () => {
    render(<MessageButton disabled={false} />);
    expect(screen.getByText(/Отправить/)).toBeInTheDocument();
  });

  it('render multiply components', () => {
    render(
      <>
        <MessageButton disabled={false} />
        <MessageButton disabled={false} />
      </>
    );
    expect(screen.queryAllByRole('button').length).toBe(2);
  });

  it('button is disabled', () => {
    render(<MessageButton disabled />);
    expect(screen.getByText('Отправить')).toBeDisabled();
  });

  it('button have style message-button', () => {
    render(<MessageButton disabled={false} />);
    expect(screen.getByText('Отправить')).toHaveClass('chat-message-button');
  });

  it('button click with userEvent', async () => {
    const mockHandler = jest.fn();

    render(<MessageButton disabled={false} onClick={mockHandler} />);
    await userEvent.click(screen.getByText(/Отправить/));
    expect(mockHandler).toBeCalledTimes(1);
  });

  it('button async click', async () => {
    const mockHandler = jest.fn();
    render(
      <MessageButton
        disabled={false}
        onClick={() => setTimeout(mockHandler, 1000)}
      />
    );

    await userEvent.click(screen.getByText(/Отправить/));

    await waitFor(() => expect(mockHandler).toHaveBeenCalledTimes(1), {
      timeout: 1100,
    });
  });
});
