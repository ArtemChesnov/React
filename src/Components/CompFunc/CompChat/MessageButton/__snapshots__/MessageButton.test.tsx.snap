// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageButton render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="chat-message-button"
    id="button"
    type="submit"
  >
    Отправить
  </button>
</DocumentFragment>
`;
